### ISSUE TITLE
Useable for Lobo Browser Cobra Engine?

### ISSUE THREAD

**Issue by xaoseric** [jhy/jsoup#563]
Hello,
I am with the Lobo Browser project. We are currently using the w3c package for our Cobra engine. Would it be possible to use your library for rendering HTML 5? Also, would you be interested in developing the Cobra engine with us?

**Comment by jhy** [comment]
Hi,

I'm not familiar with your project to say if jsoup would be suitable. Obviously you're more than welcome to use the library under the terms of the MIT license. The concept sounds interesting and if there are changes we can make to jsoup to help the integration that would be useful for others, would certainly look at that.

Do you have any existing pointers or resources? I did a quick search but only saw some old / dormant pages.

One issue that I expect would make jsoup not suitable for use in a browser is that it doesn't currently support stream processing or incremental DOM delivery. I would expect that to be critical for performance in a browser.

**Comment by xaoseric** [comment]
Hi there, 
Thank you for the approval. There is documentation at http://oswetto.github.io/Loboevolution/. The parser/renderer engine Cobra does have the stream processing which we will implement into jsoup as well which actually is a plus for jsoup as well. The structure of the packages is undergoing a name scheme change the cobra engine can be found at https://github.com/lobobrowser/Cobra.

In addition to adding html 5 support into cobras using jsoup, we will also add features to jsoup that make it more easy for developers to leverage both the speed of jsoup with the api of cobra. Essentially, developers could use cobra and get access to all the rendering api - buttons, html, or developers could choose to leverage jsoup and get access to a Dom reader.

**Comment by jhy** [comment]
I had a bit of a poke around and saw a lot of GPL and different copyright
holders. Then your new project is licensed as MIT and has a different
copyright owner. How does that work?

**Comment by xaoseric** [comment]
Hi there,
We recently moved from GPL to MIT licensing, and as such we are updating the code base. Before, we were using the org.w3c package for html dom handling, however it is time that we use a better solution. and jsoup provides the foundations for handling html5 hence why we have decided to drop support for the w3c package. I have exchanged emails with the old lead developer, he approved the transition to MIT, and he is the one that transferred the project to us on sourceforge.The official project page on sourceforge is located at http://sourceforge.net/p/xamj, We will also be adding in support for the okhttp package. We are also working on migrating from the Swing ui to the JavaFX UI, so basically, the project is undergoing a massive rewrite from the ground up with new concepts and ui. Hence why we are able to move from GPL to MIT.
